scipy/interpolate/_rgi.py

```diff
@@ -268,9 +268,15 @@ def __call__(self, xi, method=None):
         xi : ndarray of shape (..., ndim)
             The coordinates to evaluate the interpolator at.
 
-        method : str
-            The method of interpolation to perform. Supported are "linear" and
-            "nearest".
+        method : str, optional
+            The method of interpolation to perform. Supported are "linear",
+            "nearest", "slinear", "cubic", "quintic" and "pchip". Default is
+            the method chosen when the interpolator was created.
+
+        Returns
+        -------
+        values_x : ndarray, shape xi.shape[:-1] + values.shape[ndim:]
```

Comment:
Was I saying already that this PR is unexpectedly effective in smoking out pre-existing issues?

Looking at the new test_nonscalar_values_2 made me realize this statement is strictly speaking incorrect? Ditto the matching statement in `interpn`, where this line come from.
Having `xi.shape == (4,)`  (a 4D interpolation) and `values.shape=(6, 7, 8, 9, 2, 3)` returns an array of shape `(1, 2, 3)`. 
This only happens for 1D `xi` arrays, as test_nonscalar_values shows: `xi.shape=(7, 11, 4)` and `values.shape=(6, 6, 6, 6, 8)` leads to the result of shape `(7, 11, 8)`, which agrees to what's written in the docstring. 

This behavior is already observed on main for linear and nearest modes; this PR makes new modes consistent, so there's nothing to fix in this PR. Documentation might be nice to update though in a follow-up PR. Not sure how to phrase this succinctly, maybe numpy wizards could suggest the right wording.

Reply:
So if I am understanding correctly the behaviour is:
if `xi.ndim == 1`:
`values_x : ndarray, 1 + values.shape[ndim:]`
if `xi.ndim > 1`:
`values_x : ndarray, shape xi.shape[:-1] + values.shape[ndim:]`

Reply:
Yes, that's my understanding as well (modulo a typo above: for 1D xi, the shape is `(1,) + values.shape[ndim:]` , with a one-element tuple.

Reply:
As far as I am aware the return type/shape needs to go on one line for it to render properly so I am not sure how this can neatly be documented?

Reply:
How about keeping this as is and adding a clarifying note in the Notes section